lib/queries.js

```diff
@@ -3,178 +3,104 @@ import { querySudo as query, updateSudo as update } from "@lblod/mu-auth-sudo";
-  		    <http://www.w3.org/ns/adms#status> <http://redpencil.data.gift/id/concept/JobStatus/success>;
-            <http://redpencil.data.gift/vocabularies/tasks/operation> ?operation;
-            <http://purl.org/dc/terms/modified> ?modified.
-            filter (?type in(<http://vocab.deri.ie/cogs#Job>,<http://vocab.deri.ie/cogs#ScheduledJob>))
-
-
-    }} order by desc(?modified) limit 1
-`;
-  const response = await query(queryStr, {}, connectionOptions);
-  if (response?.results?.bindings?.length) {
-    return new Date(response.results.bindings[0].modified.value);
-  } else {
-    console.log("no last dump file found");
-    return new Date(0); // we keep everything (Since 1970 ;-P)
-  }
+  console.log("Dump file logic is deprecated - using date-based retention only");
+  return new Date(0); // we keep everything based on date only (Since 1970 ;-P)
 }
 
-function operationValues(operations) {
+function jobOperationFilter(operations) {
   if (operations.length === 0) {
     return '';
   } else {
-    return `VALUES ?operation {${operations
+    return `VALUES ?jobOperation {${operations
       .map((o) => sparqlEscapeUri(o))
       .join(" ")}}`;
   }
 }
 
-function cleanupUrl(u) {
-  let url = new URL(u);
-  url.pathname = "";
-  url.search = "";
-  return url.toString();
-}
-
 export async function getJobWithStatusAndBeforeDate(status, date) {
-  const q = `select distinct ?job where {
-    graph <${DEFAULT_GRAPH}> {
-      ?job a ?type;
-           <http://www.w3.org/ns/adms#status> <${status}>;
-           <http://purl.org/dc/terms/modified> ?modified.
-      filter (?modified < ${sparqlEscapeDateTime(date)} && ?type in(<http://vocab.deri.ie/cogs#Job>,<http://vocab.deri.ie/cogs#ScheduledJob>))
+  const q = `
+    PREFIX task: <http://redpencil.data.gift/vocabularies/tasks/>
+    PREFIX dct: <http://purl.org/dc/terms/>
+    PREFIX adms: <http://www.w3.org/ns/adms#>
+    PREFIX cogs: <http://vocab.deri.ie/cogs#>
 
-    }
+    SELECT DISTINCT ?job WHERE {
+      ${jobOperationFilter(JOB_OPERATIONS)}
+      GRAPH <${DEFAULT_GRAPH}> {
+        ?job a ?type;
+             adms:status ${sparqlEscapeUri(status)};
+             dct:modified ?modified.
+        FILTER (?modified < ${sparqlEscapeDateTime(date)} && ?type IN (cogs:Job, cogs:ScheduledJob))
 
-  }`;
+        ${JOB_OPERATIONS.length > 0 ? '?job task:operation ?jobOperation.' : ''}
+      }
+    }`;
   let res = await query(q, {}, connectionOptions);
   return res.results.bindings.map((r) => r.job.value);
 }
-export async function getSuccessfulJobsBeforeDate(date) {
-  const jobsToClean = [];
-  let maxDaysToKeepSuccessFulJobs = new Date();
-  maxDaysToKeepSuccessFulJobs.setDate(
-    maxDaysToKeepSuccessFulJobs.getDate() - MAX_DAYS_TO_KEEP_SUCCESSFUL_JOBS,
+export async function getSuccessfulJobsBeforeDate() {
+  const maxDaysToKeepSuccessfulJobs = new Date();
+  maxDaysToKeepSuccessfulJobs.setDate(
+    maxDaysToKeepSuccessfulJobs.getDate() - MAX_DAYS_TO_KEEP_SUCCESSFUL_JOBS,
```

Comment:
I feel like this date calculation should be done when calling the function, so not in this `queries.js` file, where we expect to see only (or mostly only) queries and supporting code for executing those queries.